pyvrp/cpp/ProblemData.cpp

```diff
@@ -104,6 +104,41 @@ ProblemData::Client::Client(Client &&client)
 
 ProblemData::Client::~Client() { delete[] name; }
 
+ProblemData::Depot::Depot(Coordinate x,
+                          Coordinate y,
+                          Duration twEarly,
+                          Duration twLate,
+                          char const *name)
+    : x(x), y(y), twEarly(twEarly), twLate(twLate), name(duplicate(name))
+{
+    if (twEarly > twLate)
+        throw std::invalid_argument("tw_early must be <= tw_late.");
+
+    if (twEarly < 0)
+        throw std::invalid_argument("tw_early must be >= 0.");
+}
+
+ProblemData::Depot::Depot(Depot const &depot)
+    : x(depot.x),
+      y(depot.y),
+      twEarly(depot.twEarly),
+      twLate(depot.twLate),
+      name(duplicate(depot.name))
+{
+}
+
+ProblemData::Depot::Depot(Depot &&depot)
+    : x(depot.x),
+      y(depot.y),
+      twEarly(depot.twEarly),
+      twLate(depot.twLate),
+      name(depot.name)  // we can steal
+{
+    depot.name = nullptr;  // stolen
```

Comment:
There is no test for this since it's fundamentally a C++ mechanic that isn't really available to test from Python, but it should be clear from reading it that this is indeed correct.